src/cloudai/_core/base_job.py

```diff
@@ -14,40 +14,121 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Union
+
+from .system import System
 from .test import Test
 
 
-class BaseJob:
+class BaseJob(ABC):
     """
     Base class for representing a job created by executing a test.
 
     Attributes
-        id (int): The unique identifier of the job.
+        mode (str): The mode of the job (e.g., 'run', 'dry-run').
+        system (System): The system in which the job is running.
```

Comment:
I don't think `Job` should know about `System`. We already link them on `Runner` level, IMO it is a good idea to keep logic there.

Let's define which object does what and will try to stick to this idea. Something like this, but more detailed (note, that some facts below might be incorrect).
1. `System` — describes a system and knows how to interact with it.
2. `Job` — keeps track of a job and its dependencies. 
3. `Runner` — accepts a System and a Job and know how to run and keep track of it.

Reply:
Yes, you are right; the base job class does not need to know about its system class. Previously, I added this to refactor the code for unifying the job spec generation interface. It turns out that we no longer need this.